flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/asyncprocessing/AbstractAsyncStateStreamOperator.java

```diff
@@ -133,6 +140,40 @@ public final <T> ThrowingConsumer<StreamRecord<T>, Exception> getRecordProcessor
                         getClass().getName(), inputId));
     }
 
+    /**
+     * Returns a {@link org.apache.flink.streaming.api.operators.InternalTimerServiceAsyncImpl} that
+     * can be used to query current processing time and event time and to set timers. An operator
+     * can have several timer services, where each has its own namespace serializer.
+     *
+     * <p>Timers are always scoped to a key, the currently active key of a keyed stream operation.
+     * When a timer fires, this key will also be set as the currently active key, and a {@link
+     * RecordContext} will also be set as the current processing context to preserve record order.
```

Comment:
It might be better to keep descriptions to implementation details as inline comments instead of as JavaDocs.